src/sparse/KokkosSparse_sptrsv_supernode.hpp

```diff
@@ -1109,46 +1170,105 @@ invert_supernodal_columns(KernelHandle kernelHandle, bool unit_diag, int nsuper,
   // quick return
   if (!invert_diag) return;
 
+  #ifdef KOKKOS_SPTRSV_SUPERNODE_PROFILE
   Kokkos::Timer timer;
   double time1 = 0.0;
   double time2 = 0.0;
+  double time3 = 0.0;
+  #endif
 
   // ----------------------------------------------------------
   // now let's invert some blocks
+  // > first go through all the supernode columns
+  // > use KokkosBlas on large blocks, and keep track of small blocks
+  // > to call batchedBlas on them
+  int num_batchs = 0;
+  int size_unblocked = handle->get_supernode_size_unblocked();
+  integer_view_host_t supernode_ids ("supernode_batch", nsuper);
   for (int s2 = 0; s2 < nsuper; s2++) {
```

Comment:
Consider experimenting with `nsuper` and determining a `parallel_switch`. If `nsuper` > `parallel_switch`, could we perform two passes:
1. parallel or serial scan `nb` for batched matrix and non-batched matrix indices.
2. apply block inversions in parallel on:
  a. batched matrices
  b. non-batched matrices
3. fence before reading the inversions

Reply:
Thank you for the review @e10harvey !! I'll try moving the if-statement out of functor.

In term of performance, if we run it with one-thread, it may run slower if batched TRTRI/TRTRM runs slower than device TRTRI/TRTRM for the small blocks on one core. We can play with the "size_unblock" which is used to decide when to switch between the batch vs. device calls. But this parameter is also used to decide when to switch during the solve. So, I may need to split the parameter to "size_unblock_setup" and "size_unblock_solve".

I am not sure if we can "apply non-batched/device calls in parallel"?

Reply:
> I am not sure if we can "apply non-batched/device calls in parallel"?

I am thinking about an effort to saturate the hardware; maybe two main host threads - one targeting the device(s) for non-batched matrices and another targeting all available host threads for batched matrices.

Reply:
Wow.. You are thinking to split the threads for calling batched kernels, and for calling KokkosBlas::trtri & KokkosBlast::trmm in sequence (these may be threaded kernels, with TPL?), in parallel? Interesting idea, I can take a look (e.g., I am not sure how many threads to use on each, and big blocks may just dominate), maybe after this PR..